lib/haml_coffee_assets/action_view/resolver.rb

```diff
@@ -8,7 +8,7 @@ module ActionView
     # in it would normally be a fallback for all formats.
     #
     class Resolver < ::ActionView::FileSystemResolver
-      def find_templates(name, prefix, partial, details)
+      def find_templates(name, prefix, partial, details, outside_app_allowed = false)
```

Comment:
@jasonfb I'm not sure that's what's happening here, although I'll admit I can't quite juggle all the monkey patching of internal methods and superchain calls. I think what we're talking about on older versions of Rails is something like this:

```ruby
class Foo
  def foo(a) # on "newer" versions of Rails this would be def foo(a, b)
    p a
  end
end

module MyLibrary
  class Bar < ::Foo
    def foo(a, b = false)
      super
    end
  end
end
```

With this, both these error:

```ruby
MyLibrary::Bar.new.foo(123) # ArgumentError: wrong number of arguments (given 2, expected 1)
MyLibrary::Bar.new.foo(123, :xyz) # ArgumentError: wrong number of arguments (given 2, expected 1)
```

`Bar#foo` has an optional argument, but the implicit `super` is going to try and pass those up no matter what, because it doesn't really "know" the value was given implicitly or explicitly.